zebra-chain/src/orchard/commitment.rs

```diff
@@ -26,14 +27,16 @@ use super::sinsemilla::*;
 /// Generates a random scalar from the scalar field 𝔽_{q_P}.
 ///
 /// <https://zips.z.cash/protocol/nu5.pdf#pallasandvesta>
-pub fn generate_trapdoor<T>(csprng: &mut T) -> pallas::Scalar
+pub fn generate_trapdoor<T>(csprng: &mut T) -> Result<pallas::Scalar, RandError>
 where
     T: RngCore + CryptoRng,
 {
     let mut bytes = [0u8; 64];
-    csprng.fill_bytes(&mut bytes);
-    // pallas::Scalar::from_bytes_wide() reduces the input modulo q_P under the hood.
-    pallas::Scalar::from_uniform_bytes(&bytes)
+    csprng
+        .try_fill_bytes(&mut bytes)
+        .map_err(|_| RandError::FillBytes)?;
+    // pallas::Scalar::from_uniform_bytes() reduces the input modulo q_P under the hood.
+    ok(pallas::Scalar::from_uniform_bytes(&bytes))
```

Comment:
Not sure how this happened, maybe autocorrect?

```suggestion
    Ok(pallas::Scalar::from_uniform_bytes(&bytes))
```